ignite/contrib/handlers/tqdm_logger.py

```diff
@@ -232,25 +233,25 @@ class _OutputHandler(BaseOutputHandler):
     def __init__(
         self,
         description: str,
-        metric_names: Optional[str] = None,
+        metric_names: Optional[Union[str, List[str]]] = None,
         output_transform: Optional[Callable] = None,
-        closing_event_name: EventEnum = Events.EPOCH_COMPLETED,
-    ):
+        closing_event_name: Events = Events.EPOCH_COMPLETED,
+    ) -> None:
         if metric_names is None and output_transform is None:
             # This helps to avoid 'Either metric_names or output_transform should be defined' of BaseOutputHandler
             metric_names = []
         super(_OutputHandler, self).__init__(description, metric_names, output_transform, global_step_transform=None)
         self.closing_event_name = closing_event_name
 
     @staticmethod
-    def get_max_number_events(event_name: Union[CallableEventWithFilter, Enum], engine: Engine):
+    def get_max_number_events(event_name: Union[str, Events], engine: Engine) -> int:
         if event_name in (Events.ITERATION_STARTED, Events.ITERATION_COMPLETED):
-            return engine.state.epoch_length
+            return cast(int, engine.state.epoch_length)
```

Comment:
I understand here mypy's reasoning as `epoch_length` and `max_epochs` can be both None (for unknown-size iterators and when user sets max_iters instead of max_epochs). Probably, we can improve it by explicitly saying that the output can be also None...

Reply:
ok, let me check, what happens 😄